src/extraction/features/stratigraphy/layer/continuation_detection.py

```diff
@@ -161,6 +178,47 @@ def _is_continuation(
     return not (depths and prev_depths and depths[0] < np.quantile(prev_depths, 1 - DEPTHS_QUANTILE_SLACK))
 
 
+def _find_optimal_split(
+    borehole_to_extend: ExtractedBorehole, borehole_continuation: ExtractedBorehole, last_duplicate_layer_index: int
+) -> tuple[int, int]:
+    """Find cut indices to remove duplicated layers when merging two boreholes.
+
+    When a borehole continues on the next page, some top layers of the new page
+    may duplicate the bottom layers of the previous page. This function walks the
+    overlap (from the bottom up) and shifts the split upward if the upper
+    layer lacks depth information while the corresponding lower layer has it.
+    This preserves the most complete layer data across the boundary.
+
+    Args:
+        borehole_to_extend (ExtractedBorehole): The borehole from the previous page.
+        borehole_continuation (ExtractedBorehole): The borehole from the current page.
+        last_duplicate_layer_index (int): The index of the last duplicated layer in the continuing borehole.
+
+    Returns:
+        tuple[int, int]: `(id_upper_end, id_lower_start)` such that concatenating these two slices removes
+        duplicates while keeping the most informative depth entries.
+    """
+    # Check the number of layer that overlaps
+    n_overlap = min(len(borehole_to_extend.predictions), last_duplicate_layer_index + 1)
+    offset = 0
+
+    # Check if threshold should be moved
+    for i in range(n_overlap):
+        upper_layer = borehole_to_extend.predictions[-(i + 1)]
+        lower_layer = borehole_continuation.predictions[last_duplicate_layer_index - i]
+        # It is possible that the information of the upper layers is cut (), check if we can retrieve information from
+        # the lower layers
+        if not upper_layer.depth_nonempty() and lower_layer.depth_nonempty():
+            offset += 1
+            continue
+        # As soon as information is missing in the lower layer, stop iteration
+        break
```

Comment:
I would prefer
```suggestion
        if not upper_layer.depth_nonempty() and lower_layer.depth_nonempty():
            offset += 1
        else:
            # As soon as information is missing in the lower layer, stop iteration
            break
```
I would minimize the use of control-flow-affecting keywords such as `continue`  and `break`, as they quickly make the logic harder to understand. In this case for example, my initial reaction was "this `break` statement will always be executed at the end of the first iteration, making the whole `for` loop quite useless", because I had not immediately spotted that there was a `continue` inside an `if` clause higher up.